### DISCUSSION TITLE
Home Page UI

### DISCUSSION THREAD

**Discussion by ghost** [bromite/bromite#1141] [Q&A] [answered]
Anyone know how to enable feeds??
I mean that news cards under sites on NTP. Or any other ways to customize the NTP on bromite??😊

**Comment by csagan5** [comment, marked as answer]
It is not possible, the feeds feature uses information from Google servers and it is disabled in Bromite.

**Reply by ghost** [reply]
Ok sir

**Reply by csagan5** [reply]
@uazo @thestinger perhaps we could allow this functionality if it is not mediated by any cloud service: https://www.theverge.com/2021/5/20/22445284/google-rss-chrome-android-feature-test

**Reply by uazo** [reply]
I don't know that code, by chance it is this? https://chromium-review.googlesource.com/q/hashtag:webfeed

flag "WebFeed": https://source.chromium.org/chromium/chromium/src/+/main:components/feed/feed_feature_list.cc;l=57;drc=9fcc9d7b2915b6192ee6810eec54c50deb6313c6;bpv=1;bpt=1

if it is that it is to be verified well because contrary to what it could be, chromium would behave in an "active" way, I have seen calls via post https://source.chromium.org/chromium/chromium/src/+/main:components/feed/core/v2/feed_network.h;l=95;drc=9fcc9d7b2915b6192ee6810eec54c50deb6313c6;bpv=1;bpt=1 however it is not clear to me if those calls are active only if gaia is active

here the relative test-case, I will check
https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/android/feed/v2/rss_links_fetcher_browsertest.cc

**Reply by csagan5** [reply]
> if it is that it is to be verified well because contrary to what it could be, chromium would behave in an "active" way, I have seen calls via post https://source.chromium.org/chromium/chromium/src/+/main:components/feed/core/v2/feed_network.h;l=95;drc=9fcc9d7b2915b6192ee6810eec54c50deb6313c6;bpv=1;bpt=1 however it is not clear to me if those calls are active only if gaia is active

If user subscribes to an RSS feed then it's fine that calls are made to fetch such feeds. The only calls to disallow are to hosts that the user did not subscribe to.

**Reply by uazo** [reply]
> If user subscribes to an RSS feed then it's fine that calls are made to fetch such feeds

the calls are to google.

I saw that gaia has nothing to do with it, that is, from a static reading of the code, the calls are there regardless of the google login.
everything is related to `BUILDFLAG(ENABLE_FEED_V2)`, which, however, he should be active for webfeeds.

the call chain appears to be
`FollowWebFeedDiscoverApi` -> `SubscribeToWebFeedTask::Run()` ->  `WebFeedSubscriptionCoordinator::FollowWebFeedFromUrlStart()` -> `WebFeedSubscriptionCoordinator::FollowWebFeed()` -> `JNI_WebFeedBridge_FollowWebFeed()` -> `WebFeedBridge` (java) but I haven't seen any `if is logged`

the only reason it wouldn't go through those calls is to act on `FeedService` technically acting on the instance generator here https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/android/feed/v2/feed_service_factory.cc;l=71;bpv=1;bpt=1 where then you get to the buildflag

but I still don't understand if the list of feeds the user has subscribed to is local or just remote, and of course, it should be checked first if those `POST` are really there

**Reply by uazo** [reply]
here the evidence that it is a call to google
https://source.chromium.org/chromium/chromium/src/+/main:components/feed/core/v2/feed_network_impl.cc;l=235
